context/context.go

```diff
@@ -20,6 +24,100 @@ type Context interface {
 	SetBaseRepo(string)
 }
 
+// cap the number of git remotes looked up, since the user might have an
+// unusally large number of git remotes
+const maxRemotesForLookup = 5
+
+func ResolveRemotesToRepos(remotes Remotes, client *api.Client, base string) (ResolvedRemotes, error) {
+	sort.Stable(remotes)
+	lenRemotesForLookup := len(remotes)
+	if lenRemotesForLookup > maxRemotesForLookup {
+		lenRemotesForLookup = maxRemotesForLookup
+	}
+
+	hasBaseOverride := base != ""
+	baseOverride := ghrepo.FromFullName(base)
+	foundBaseOverride := false
```

Comment:
i'm not that familiar with golang types, do they have optionals or nullables? it feels like baseOverride could be "string or nah", and we could eliminate `hasBaseOverride` or maybe the logic flow could go

> if (base !== '') attempt to add base override repo to repos list
> else get resolved remotes

the main reason i'm suggesting this is it took me awhile to tease apart the different pieces of control logic in this function

Reply:
there is not an optional type D: this is just kind of how golang would have us do unfortunately.